tests/test_filters.py

```diff
@@ -0,0 +1,137 @@
+            2, page__node__site=site1, language="en"
+        )
+        site2_pagecontents = [
+            PageContentWithVersionFactory(page__node__site=site2, language="en")
+        ]
+        model = PageContent
+        url = self.get_admin_url(model, "changelist")
+        with self.login_user_context(self.get_superuser()):
+            with self.settings(SITE_ID=site1.pk):
+                response1 = self.client.get(url)
+            with self.settings(SITE_ID=site2.pk):
+                response2 = self.client.get(url)
+        self.assertEqual(response1.status_code, 200)
+        self.assertEqual(response2.status_code, 200)
+        self.assertEqual(set(site1_pagecontents), set(response1.context["cl"].queryset))
+        self.assertEqual(set(site2_pagecontents), set(response2.context["cl"].queryset))
+
+
+class LanguageFilterTestCase(CMSTestCase):
+    def test_language_filter(self):
+        expected_en = PageContentWithVersionFactory.create_batch(3, language="en")
+        expected_de = PageContentWithVersionFactory.create_batch(3, language="de")
+        model = PageContent
+        base_url = self.get_admin_url(model, "changelist")
+        with self.login_user_context(self.get_superuser()):
+            # en is the default language configured for the site
+            response_default = self.client.get(base_url)
+            qs_default = response_default.context["cl"].queryset
+            response_en = self.client.get(base_url + "?language=en")
+            qs_en = response_en.context["cl"].queryset
+            response_de = self.client.get(base_url + "?language=de")
+            qs_de = response_de.context["cl"].queryset
+
+        self.assertEqual(set(qs_default), set(expected_en))
+        self.assertEqual(set(qs_en), set(expected_en))
+        self.assertEqual(set(qs_de), set(expected_de))
+
+
+class UnpublishedTestCase(CMSTestCase):
+    def test_unpublished_filter(self):
+        expected = PageContentWithVersionFactory.create_batch(3, language="en")
+        expected_unpublished = PageContentWithVersionFactory.create_batch(
+            2, language="en", version__state=UNPUBLISHED
+        )
+        model = PageContent
+        base_url = self.get_admin_url(model, "changelist")
+        with self.login_user_context(self.get_superuser()):
+            # en is the default language configured for the site
+            response_default = self.client.get(base_url)
+            qs_default = response_default.context["cl"].queryset
+            response_unpublished = self.client.get(base_url + "?unpublished=1")
+            qs_unpublished = response_unpublished.context["cl"].queryset
+
+        self.assertEqual(set(qs_default), set(expected))
+        self.assertEqual(set(qs_unpublished), set(expected_unpublished))
+
+
+class TemplateFilterTestCase(CMSTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        from django.contrib.sites.models import Site
```

Comment:
Should this be imported at the top of the file?

Reply:
Done